src/shared/components/input/input.component.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useDebouncedCallback } from 'use-debounce';
+
+import { useDispatch } from 'react-redux';
+import { FilterType } from '../../../modules/filters/filters.types';
+import { setSearch } from '../../../modules/filters/filters.actions';
+import { Container, SearchIcon, InputWrapper, InputComponent, InputUnderline } from './input.styles';
+import messages from './input.messages';
+
+export const Input = () => {
+  const dispatch = useDispatch();
+
+  const debouncedOnTextChange = useDebouncedCallback((option: FilterType) => {
+    dispatch(setSearch(option));
+  }, 500);
```

Comment:
Let's use `500` as a constant. It's not a wrong case here - but in general it's good to name timings and other constants for other developers which can easily understand what the timing means.